torch/_inductor/freezing.py

```diff
@@ -123,6 +123,15 @@ def run_node(self, node):
 
         # TODO - remove constant from node_replacement when it has no uses
         if node.op != "get_attr" and isinstance(out, torch.Tensor):
+            if (
```

Comment:
does constant folding have a way to configure pure/impure ops?

Reply:
Sorry, what's the means of pure/impure ops here? Is it something like the black list or white list?

Reply:
pure ops means ops that doesn't have side effect, so they can be constant folded,
impure ops means ops that has side effect and can not be constant folded, I remember Torchscript or maybe fx has a way to mark the op as impure and constant folding will stop at the impure ops

Reply:
Great suggestions. I found `impure` is used in fx to eliminate deadcode and generate `FoldedGraphModule`:

- https://github.com/pytorch/pytorch/blob/07107919297db3f8ab37f11c12666b6d6d5f692e/torch/fx/graph.py#L1427
- https://github.com/pytorch/pytorch/blob/07107919297db3f8ab37f11c12666b6d6d5f692e/torch/fx/experimental/const_fold.py#L181

Here, we also add the `is_impure` check which only contains check of `dequantize_per_channel` for now.